merkletree/proof.go

```diff
@@ -80,38 +80,51 @@ func (ap *AuthenticationPath) authPathHash() []byte {
 	return hash
 }
 
-func (ap *AuthenticationPath) verifyBinding(key, value []byte) bool {
+// VerifyBinding verifies both the value and the commitment
+// of the proof node of the authentication path.
+//
+// This should be called only when ap is a proof of inclusion.
+func (ap *AuthenticationPath) VerifyBinding(key, value []byte) bool {
```

Comment:
Yes, since `ap.proofType` is unexported, it shouldn't be transmitted over the wire. So setting `proofType` from the server-side has no impact on the client.